lib/rubocop/cop/style/disable_cop_for_entire_file_in_source_code.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # rubocop:disable Lint/RedundantCopDisableDirective
+    module Style
+      # Enforces that disabling cops for an entire file happens in a config
+      # file, rather than by wrapping the entire file in `rubocop:disable` and
+      # `rubocop:enable` comments.
+      #
+      # @example
+      #   # bad
+      #   # rubocop:disable Department/CopName
+      #   entire_file
+      #   # rubocop:enable Department/CopName
+
+      #   # bad
+      #   # rubocop:todo Department/CopName
+      #   entire_file
+      #   # rubocop:enable Department/CopName
+      #
+      #   # good
+      #   # Department/CopName disabled in config, such as .rubocop.yml or .rubocop_todo.yml
+      #   entire_file
+      #
+      #   # good
+      #   code
+      #   # rubocop:disable Department/CopName
+      #   subset_of_code
+      #   # rubocop:enable Department/CopName
+      #   more_code
+      class DisableCopForEntireFileInSourceCode < Base
+        # rubocop:enable Lint/RedundantCopDisableDirective
+        MSG = 'When disabling cops for an entire file, prefer excluding the file in `.rubocop.yml`.'
+        TODO_MSG = 'When marking cops as TODO for an entire file, ' \
+                   'prefer excluding the file in `.rubocop_todo.yml`.'
```

Comment:
I should update these messages to also suggest approaches in descending order of preference:
1. Disable the cop(s) per line
  _See the change made to `lib/rubocop/cop/style/block_delimiters.rb`._
3. Disable the cop(s) for only the affected range
  _See the change made to `lib/rubocop/cop/lint/redundant_cop_enable_directive.rb`._
5. Disable the cop in a config file (ideally only if to remove friction from expecting to add future offenses)
  _I considered this for `spec/rubocop/cop/lint/script_permission_spec.rb`, but ended up using in-line disabling, since there were only two occurrences._

Reply:
How would you determine which option to show, or are you suggesting showing all of them? If so, it'd probably be better served in the documentation for the cop and keeping the message more terse.

Reply:
Hmm... Good point. I guess my main thing is that I think authors should first ask themselves "can I be more specific when disabling this cop?" before just moving it to the config file.

Reply:
Perhaps something along these lines...

```ruby
MSG = 'Prefer using directives on smaller sections of code, or if you need to disable the entire file, do it in your configuration file.'
```